### PR TITLE
unit tests and linting on python 3.12

### PR DESCRIPTION
**Pull request by dariuszniedzielski-codilime**
# Pull Request summary:
support python 3.12 

# Description of changes:
bumped required python to ^3.8.1
bumped required flake8 to ^6.1.0 (this is only dev dependency)
omit double-quote checks for python 3.12 linting job (breaking changes in f-strings)

# Checklist:
- [x] Make sure to run pre-commit before committing changes
- [x] Make sure all checks have passed
- [x] PR description is clear and comprehensive
- [x] Mentioned the issue that this PR solves (if applicable)
- [x] Make sure you test the changes

### CODE REVIEW

**Review by sbasan — APPROVED** [review]
lgtm

File: vmngclient/session.py [hunk 1]
```diff
@@ -246,7 +246,7 @@ def __create_base_url(self) -> str:
         scheme: str = url.scheme or "https"
         base_url = urlunparse((scheme, netloc, "", None, None, None))
         if self.port:
-            return f"{base_url}:{self.port}"
+            return f"{base_url}: {self.port}"
```
Comment (sbasan) [inline vmngclient/session.py:249]: Wait...
Isn't that breaking anything?
url will now contain space before port: `127.0.0.1 :433`

**Review by sbasan — CHANGES_REQUESTED** [review]
suspicious change in url generation

File: vmngclient/session.py (hunk 1 above)
Comment (dariuszniedzielski-codilime) [reply vmngclient/session.py]: You are right. Fix already committed

**Review by sbasan — APPROVED** [review]
lgtm

**Comment by sbasan** [comment]
Bumping flake8 to ^6.1.0 requires bumping required python to ^3.8.1 (even though flake8 is dev only dependency)
If everyone is ok with that we can merge this PR

**Review by sbasan — APPROVED** [review]
lgtm

**Review by cicharka — APPROVED** [review]

**Pull request merged**